src/library_wasmfs_opfs.js

```diff
@@ -283,15 +363,24 @@ mergeInto(LibraryManager.library, {
     }
 
     {{{ makeSetValue('nreadPtr', 0, 'nread', 'i32') }}};
-    _emscripten_proxy_finish(ctx);
+    wasmfsOPFSProxyFinish(ctx);
   },
 
+  _wasmfs_opfs_write_access__sig: 'iipii',
   _wasmfs_opfs_write_access__deps: ['$wasmfsOPFSAccessHandles'],
+#if USE_PTHREADS
   _wasmfs_opfs_write_access: function(accessID, bufPtr, len, pos) {
+#else
+  _wasmfs_opfs_write_access: async function(accessID, bufPtr, len, pos) {
```

Comment:
I'm thinking we should add a preprocessor that is something like `{{{maybeAsync}}}` where with asyncify enabled it would be converted to `async` or empty otherwise. Similarly, I'd add `{{{maybeAwait}}}`.

Reply:
That sounds like good idea yes.